metagraph/src/graph/representation/base/sequence_graph.cpp

```diff
@@ -80,9 +80,33 @@ bool DeBruijnGraph::find(std::string_view sequence,
     return num_kmers_missing <= max_kmers_missing;
 }
 
-bool DeBruijnGraph::operator==(const DeBruijnGraph &) const {
-    throw std::runtime_error("Not implemented");
-    return false;
+void DeBruijnGraph
+::adjacent_outgoing_nodes(node_index node,
+                          const std::function<void(node_index)> &callback) const {
+    assert(node > 0 && node <= max_index());
+    call_outgoing_kmers(node, [&](auto child, char) { callback(child); });
+}
+
+void DeBruijnGraph
+::adjacent_incoming_nodes(node_index node,
+                          const std::function<void(node_index)> &callback) const {
+    assert(node > 0 && node <= max_index());
```

Comment:
Using `max_index()` should fix unit test failure on DBGHashFast, but is it intended that they are different?

Reply:
yes, if there's a graph without a contiguous index space. we do require, however, than `max_index() >= num_nodes()`